packages/expo-updates/android/src/main/java/expo/modules/updates/loader/FileDownloader.java

```diff
@@ -55,28 +55,21 @@
     void onSuccess(AssetEntity assetEntity, boolean isNew);
   }
 
-  private static OkHttpClient getClient(Context context) {
-    if (sClient == null) {
-      sClient = new OkHttpClient.Builder().cache(getCache(context)).build();
-    } else {
-      if (sClient.cache() == null || !getCacheDirectory(context).getAbsolutePath().equals(sClient.cache().directory().getAbsolutePath())) {
-        throw new AssertionError("Error: trying to access static OkHttpClient that was created with a different context.");
-      }
-    }
-    return sClient;
+  public FileDownloader(Context context) {
+    mClient = new OkHttpClient.Builder().cache(getCache(context)).build();
   }
 
-  private static Cache getCache(Context context) {
+  private Cache getCache(Context context) {
     int cacheSize = 50 * 1024 * 1024; // 50 MiB
     return new Cache(getCacheDirectory(context), cacheSize);
   }
 
-  private static File getCacheDirectory(Context context) {
-    return new File(context.getCacheDir(), "okhttp");
+  private File getCacheDirectory(Context context) {
+    return new File(context.getApplicationContext().getCacheDir(), "okhttp");
```

Comment:
Think it's worth adding a comment here to explain why the application context is used. This seems like a question someone would not be able to answer by reading this file.

Reply:
oh, good catch. this was from a previous attempted fix that didn't work. i've reverted this change.